src/components/Images.vue

```diff
@@ -298,30 +308,65 @@ export default {
     deleteImage(obj) {
       this.currentCommand = `delete ${ obj.imageName }:${ obj.tag }`;
       this.mainWindowScroll = this.$refs.fullWindow.parentElement.parentElement.scrollTop;
+      this.postOpSuccessHandler = this.postDeleteSuccessHandler;
       this.startRunningCommand('delete');
       ipcRenderer.send('confirm-do-image-deletion', obj.imageName.trim(), obj.imageID.trim());
     },
+    postDeleteSuccessHandler() {
+      if (this.imageManagerOutput === '') {
+        this.closeOutputWindow(null);
+      }
+    },
     doPush(obj) {
       this.currentCommand = `push ${ obj.imageName }:${ obj.tag }`;
+      this.mainWindowScroll = this.$refs.fullWindow.parentElement.parentElement.scrollTop;
       this.startRunningCommand('push');
       ipcRenderer.send('do-image-push', obj.imageName.trim(), obj.imageID.trim(), obj.tag.trim());
     },
     doBuildAnImage() {
       this.currentCommand = `build ${ this.imageToBuild }`;
       this.fieldToClear = 'imageToBuild';
+      this.postCloseOutputWindowHandler = this.scrollToPulledImage(this.imageToBuild);
       this.startRunningCommand('build');
       ipcRenderer.send('do-image-build', this.imageToBuild.trim());
     },
     doPullAnImage() {
       this.currentCommand = `pull ${ this.imageToPull }`;
       this.fieldToClear = 'imageToPull';
+      this.postCloseOutputWindowHandler = this.scrollToPulledImage(this.imageToPull);
       this.startRunningCommand('pull');
       ipcRenderer.send('do-image-pull', this.imageToPull.trim());
     },
+    scrollToPulledImage(imageToPull) {
+      return () => {
+        if (!this.imageManagerOutput.trimStart().startsWith('Error:')) {
+          //  error on pull: go back stay here
+          const [imageName, possibleTag] = imageToPull.split(':', 2);
+          const tag = possibleTag || 'latest';
+          const rows = document.querySelector('table.sortable-table').rows;
+
+          // Ignore header row, start at 1
+          for (let i = 1; i < rows.length; i++) {
+            const row = rows[i];
+            const children = row.children;
+
+            if (children[0].textContent.trim() === imageName && children[1].textContent.trim() === tag) {
```

Comment:
Why not use the `data-node-id` attribute, which is the key of the row?
```typescript
const image = this.images.find(image => image.imageName === imageName && image.tag === tag);
const row = this.$refs['images-table'].querySelector(`[data-node-id="${image.imageID}"]`);
```

Reply:
Most of the stuff here is better than the long loops, but I couldn't get `this.$refs['images-table']` to work on the `SortableTable`.  I don't know why, and googling isn't finding an answer, but I suspect you can't have a ref on a vue component, because they don't exist with all their attributes in the final DOM.  Wrapping the SortableTable in a div with an ID works fine.

Reply:
Odd, `this.$refs['images-table'].$el.querySelector(…)` seemed to work for me. Note that `this.$refs[…]` was the Vue component for me (not the element), so I needed the `$el`.